include/Ref.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <memory>
+
+namespace arch {
+
+template<typename T>
+using Ref = std::shared_ptr<T>;
+
+template<typename T, typename... Args>
+constexpr Ref<T> createRef(Args&&... args) {
+	return std::make_shared<T>(std::forward<Args>(args)...);
+}
+
+} // namespace arch
```

Comment:
newline

Reply:
sory, this file, shouldn't be in this PR